src/components/Form/Form.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { useForm, FormProvider } from "react-hook-form";
+
+export type FormValues = {
+  description: string;
+  agreeToTerms: boolean;
+};
+
+interface FormProps {
+  submit: (data: FormValues) => void;
+  children?: React.ReactNode;
+}
+
+const Form: React.FC<FormProps> = ({ submit, children }) => {
+  const methods = useForm<FormValues>({
+    mode: "onBlur",
+  });
+
+  const onSubmit = methods.handleSubmit((data) => {
+    console.log("Form submitted with data:", data);
+    submit(data);
+    methods.reset(); // Reset form state after submission
+    console.log("Form reset.");
+  });
+
+  return (
+    <FormProvider {...methods}>
+      <form
```

Comment:
I don't believe that the Form should look like this (attached in the image) 
<img width="643" alt="image" src="https://github.com/user-attachments/assets/b23b97f1-2d1e-48c9-b88a-cad3a33d8af1">
I believe that you would only define properties such as `flex` `flex-col` and `gap` since it is not the modal, but all input elements are children of the form component. Here is an image of how I built the Form component that I use for the parent element for my input elements (as of now at least) 
<img width="643" alt="image" src="https://github.com/user-attachments/assets/adce5abb-5949-4504-80b1-560fd7d970e5">

Reply:
got it , so just want to make sure does this look correct to you?
<img width="1440" alt="Screenshot 2024-09-12 at 4 42 49 PM" src="https://github.com/user-attachments/assets/65fc4f76-eaf9-4959-9365-6a24e20f556e">

I'll go ahead and change my textareainput and checkboxinput form's as well. 

So as of right now, I'm planning to keep the Form component with minimal styling (like flex, flex-col, and gap) to ensure it's reusable and flexible. Then, when I create the Modal component, I plan to add specific styles such as bg-wheat, p-[28px], rounded-[40px], and max-w-[467px] to define the visual appearance. Does this approach align with what you're suggesting, or would you recommend a different strategy?

Reply:
Ahh I see got it thank you for the explanation, makes a lot more sense now.